input/docs/extending/modules/best-practices.md

```diff
@@ -0,0 +1,200 @@
+> Depending on the module it might be required to target additional platforms.
+
+**_§2.6_** **Do not** include additional targets such as `net461` and `net5.0`.
+
+> **Why?** Due to current limitations ([1](https://github.com/cake-build/cake/issues/2256), [2](https://github.com/cake-build/cake/issues/2525)) on how modules are loaded, Cake attempts to load all assemblies of all targets which causes warnings to be displayed and the impression that the module was not loaded and/or is not working.
+> Once those limitations are fixed, we'll update this recommendation to align with the similar recommendations for [Cake Addins](/docs/extending/addins/best-practices#net-target-version).
+
+# Package metadata
+
+## Icons
+
+**_§3.1_** **Do** define an icon for the NuGet package.
+
+> **Why?** Icons helps the user to identify Cake modules.
+
+----------------------------------------------------------------------------------------------------
+
+**_§3.2_** **Consider** using the [Cake Contrib icon](https://github.com/cake-contrib/graphics/blob/master/png/cake-contrib-medium.png).
+
+> **Why?** When the module doesn't have an own product icon, using the [Cake Contrib icon](https://github.com/cake-contrib/graphics/blob/master/png/cake-contrib-medium.png)
+> gives a common branding to Cake modules.
+
+----------------------------------------------------------------------------------------------------
+
+**_§3.3_** **Avoid** using the Cake icon.
+
+> **Why?** The Cake icon is reserved for core Cake packages.
+
+----------------------------------------------------------------------------------------------------
+
+**_§3.4_** **Do** embed the icon in the package.
+
+> **Why?** Using an icon embedded in the package is more reliable then linking to an icon an external web site which hosts the icon.
+
+> **Example:**
+>
+> This can be done with the following line in the modules `.csproj`:
+>
+> ```xml
+> <PackageIcon>PackageIcon.png</PackageIcon>
+> ```
+>
+> The modules `.csproj` should also contain a reference to the `png` file:
+>
+> ```xml
+> <ItemGroup>
+>   <None Include="..\PackageIcon.png" Pack="true" PackagePath="" />
+> </ItemGroup>
+> ```
+>
+> Notice the `Pack` attribute, this is particularly important to ensure the file is embedded in the NuGet package.
+
+:::{.alert .alert-info}
+Until early 2019, the recommendation was to reference the Cake Contrib icon hosted on the rawgit CDN but rawgit announced that it would shutdown in October 2019 therefore the recommendation changed to reference the Cake Contrib icon hosted on the jsDelivr CDN.
+This recommendation changed once again in the fall of 2019 when NuGet started supporting embedded icons.
+:::
+
+# Documentation
+
+**_§4.1_** **Do** follow the [Cake Documentation Guidelines](/community/contributing/documentation).
+
+> **Why?** Proper XML comments and usage of [CakeAliasCategoryAttribute](/api/cake.core.annotations/cakealiascategoryattribute/)
```

Comment:
`CakeAliasCategoryAttribute` is not used for modules

Reply:
However, we should mention the [CakeModuleAttribute](https://cakebuild.net/api/Cake.Core.Annotations/CakeModuleAttribute/)

Reply:
> `CakeAliasCategoryAttribute` is not used for modules

:white_check_mark: Fixed. Thanks!

> However, we should mention the [CakeModuleAttribute](https://cakebuild.net/api/Cake.Core.Annotations/CakeModuleAttribute/)

:white_check_mark: Fixed. Thanks!